README.md

```diff
@@ -1 +1,235 @@
 # Immutable Web Apps
+
+## Introduction
+
+_Immutable Web Applications_ is a framework-agnostic methodology for building and deploying static [single-page applications](https://en.wikipedia.org/wiki/Single-page_application) that:
+
+- Minimizes risk and complexity of live releases.
+- Simplifies and maximizes caching.
+- Minimizes the need for servers and administration of runtime environments.
+- Enables continuous delivery through simple, flexible, atomic deployments.
+
+## Principles
+
+The methodology is based on the principles of ___strictly separating___:
+
+- Configuration from code.
+- Release tasks from build tasks.
+- Dynamic content from static content.
+
+These principles are inspired by [The Twelve-Factor App](https://12factor.net/).
+
+## Concepts
+
+The following concepts define the core requirements for an _Immutable Web Application_. They are framework and infrastructure agnostic.
+
+### Static assets are independent of the web application environment(s)
+
+Static assets are the files (javascript, css, images) that are generated from a build of a web application codebase. They become _immutable_ when they do not contain anything environment-specific and they are published to a unique, independent location.
+
+#### Static assets must not contain anything that is environment-specific
+
+All of the leading application frameworks ([Angular CLI](https://github.com/angular/angular-cli/wiki/stories-application-environments), [Create React App](https://cli.vuejs.org/guide/mode-and-env.html#using-env-variables-in-client-side-code), [Ember CLI](https://ember-cli.com/user-guide/#Environments), [Vue CLI 3](https://cli.vuejs.org/guide/mode-and-env.html#using-env-variables-in-client-side-code)) recommend defining environment _values_ at _compile time_. This practice requires that the static assets are generated for each environment and regenerated for any change to an environment.
+
+_Immutable Web Applications_ reference environment _variables_ that are  defined on the global scope and that are set to values in the `index.html` that are unique to the deployment:
+
+```diff
+  import { Injectable } from '@angular/core';
+  import { HttpClient } from '@angular/common/http';
+
+- // remove environment constants that are defined in the codebase or at compile time
+- import { environment } from './environments/environment';
+
+  @Injectable()
+  export class DataService {
+    configUrl = ;
+
+    constructor(
+        private http: HttpClient
+        private
+        ) { }
+
+    getData() {
+-      return this.http.get(`${environment.API}/assets/data.json`);
++      // use globally scoped environment variables that are unique to the deployment
++      return this.http.get(`${window.env.API}/assets/data.json`);
```

Comment:
Just an FYI that this will likely get some blowback from Angular people.  Since Angular is built to be somewhat platform agnostic, referencing window directly in anything but one of the top level components or app bootstrapper will make some people twitch.

Reply:
I will work on a different example that uses `environment.js` for v2